pkg/util/iptables/iptables.go

```diff
@@ -617,7 +617,7 @@ const (
 	opCreateChain operation = "-N"
 	opFlushChain  operation = "-F"
 	opDeleteChain operation = "-X"
-	opListChain   operation = "-L"
+	opListChain   operation = "-L -n"
```

Comment:
This won't work... it will pass "`-L -n`" as a single argument, not as two separate arguments.

If you look at [the logs](https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/92860/pull-kubernetes-e2e-gce/1280466272315772929/artifacts/e2e-3450d3ab9d-674b9-minion-group-21s9/kube-proxy.log) you'll see that it repeatedly claims that the canary was deleted and the iptables rules need to be recreated... because it mistakes the "syntax error" status code from iptables as meaning "could not list the chain because it doesn't exist".

We should be able to just pass "`-n`" unconditionally (ie, add it in `runContext` when constructing `fullArgs`)

Reply:
> We should be able to just pass "-n" unconditionally (ie, add it in runContext when constructing fullArgs)

it fails

I0707 15:53:41.987391       1 bounded_frequency_runner.go:279] sync-runner: retrying in 59.999931677s
I0707 15:53:41.987600       1 proxier.go:845] Syncing iptables rules
E0707 15:53:41.989259       1 proxier.go:858] Failed to ensure that filter chain KUBE-EXTERNAL-SERVICES exists: error creating chain "KUBE-EXTERNAL-SERVICES": exit status 2: iptables v1.8.3 (legacy): Illegal option `-n' with this command